prow/sidecar/run_test.go

```diff
@@ -528,3 +535,74 @@ func TestLogReaders(t *testing.T) {
 	}
 
 }
+
+func TestSideCarLogsUpload(t *testing.T) {
+	logFile, err := LogSetup()
+	if err != nil {
+		t.Fatalf("Unable to set up log file")
+	}
+	testString := "Testing...Hello world!"
+	logrus.Info(testString)
+	var once sync.Once
+
+	localOutputDir, err := ioutil.TempDir("", "testdir")
+	if err != nil {
+		t.Errorf("Unable to create temp dir: %v", err)
+	}
+	options := Options{
+		GcsOptions: &gcsupload.Options{
+			GCSConfiguration: &prowapi.GCSConfiguration{
+				PathStrategy:   prowapi.PathStrategyExplicit,
+				Bucket:         "bucket",
+				LocalOutputDir: localOutputDir,
+			},
+		},
+	}
+
+	spec := &downwardapi.JobSpec{
+		Job:  "job",
+		Type: prowapi.PostsubmitJob,
+		Refs: &prowapi.Refs{
+			Org:  "org",
+			Repo: "repo",
+			Pulls: []prowapi.Pull{
+				{
+					Number: 1,
+				},
+			},
+		},
+		BuildID: "build",
+	}
+
+	entries := options.entries()
+	metadata := combineMetadata(entries)
+	buildLogs := logReaders(entries)
+
+	options.doUpload(context.Background(), spec, true, false, metadata, buildLogs, logFile, &once)
+
+	files, err := ioutil.ReadDir(localOutputDir)
+	if err != nil {
+		t.Errorf("Unable to access files in directory: %v", err)
+	}
+
+	var s []byte
+	var found bool
+	for _, file := range files {
+		if file.Name() == LogFileName {
+			found = true
+			s, err = ioutil.ReadFile(filepath.Join(localOutputDir, file.Name()))
+			if err != nil {
+				t.Fatalf("Unable to read log file: %v", err)
+			}
+		}
+	}
+	if !found {
+		t.Fatal("Log file not uploaded")
+	}
```

Comment:
Rather than searching through the directory I think we can just check for the file directly thanks to the deterministic file name `LogFileName`.
Use `filepath.Join(localOutputDir, LogFileName)`.